src/protocol/ldap/org/apache/jmeter/protocol/ldap/sampler/TrustAllSocket.java

```diff
@@ -0,0 +1,117 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.jmeter.protocol.ldap.sampler;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.net.SocketFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TrustAllSocket extends SocketFactory {
+    private static final AtomicReference<TrustAllSocket> defaultFactory = new AtomicReference<>();
+
+    private static final Logger log = LoggerFactory.getLogger(TrustAllSocket.class);
+
+    private static SSLSocketFactory sf;
+
+    public TrustAllSocket() throws NoSuchAlgorithmException, KeyManagementException {
+        TrustManager[] trustAllCerts = new TrustManager[] {new X509ExtendedTrustManager() {
+            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+            public void checkClientTrusted(X509Certificate[] certs, String authType) {
+            }
+            public void checkServerTrusted(X509Certificate[] certs, String authType) {
+            }
+            @Override
+            public void checkClientTrusted(X509Certificate[] arg0, String arg1, Socket arg2)
+                throws CertificateException {
+            }
+            @Override
+            public void checkClientTrusted(X509Certificate[] arg0, String arg1, SSLEngine arg2)
+                throws CertificateException {
+            }
+            @Override
+            public void checkServerTrusted(X509Certificate[] arg0, String arg1, Socket arg2)
+                throws CertificateException {
+            }
+            @Override
+            public void checkServerTrusted(X509Certificate[] arg0, String arg1, SSLEngine arg2)
+                throws CertificateException {
+            }
+        }};
+
+        SSLContext ctx = null;
+        ctx = SSLContext.getInstance("TLS");
+        ctx.init(null, trustAllCerts, new java.security.SecureRandom());
+        sf = ctx.getSocketFactory();
+    }
+
+    public static SocketFactory getDefault() {
+        final TrustAllSocket value = defaultFactory.get();
+        if (value == null) {
+            try {
+                defaultFactory.compareAndSet(null, new TrustAllSocket());
+            } catch (KeyManagementException e) {
+                log.error("KeyManagementException: "+e.getLocalizedMessage());
```

Comment:
Would passing the exception into the logger be best, and maybe a more descriptive error message?
```suggestion
                log.error("KeyManagementException while trying to ... ", e);
```

Reply:
honestly I don't think this error will ever happen as the custom trust manager doesn't do anything that will generate the error. Its just part of the interface. The only way the NoSuchAlgorithmException will be thrown is if the JDK does not support some version of TLS. The KeyManagementException would only be thrown in the event there is an error dealing initializing the custom SSLContext object. which is only dependent on the inputs from the custom trust manager which basically does nothing and the SecureRandom() function, which should always return some value. Both of which would be an indication of a broken JDK/JRE or a really old version.
I suppose I could provide a nicer message indicating some context of what we are trying to do here in the event an error occurs and someone sends the output to us for analysis. 
I guess something along the lines of "KeyManagementException when changing SSLContext for TrustAll" or something like that. That may help someone who isn't programmatically inclined understand what is wrong.